lib/transformer.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const zlib = require('zlib')
+const aws = require('aws-sdk')
+const {
+  checkFileFormat
+, getProperty
+, hasProperty
+} = require('./utils.js')
+
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+module.exports = {
+  extractData
+, getLogs
+, prepareLogs
+}
+
+function extractData(data) {
+  if (!data) return []
+
+  const {
+    Records
+  , logFiles
+  , ...meta
+  } = data
+
+  let ts = Date.now()
+  if (hasProperty(meta, 'eventTime')) {
+    ts = (new Date(getProperty(meta, 'eventTime'))).getTime()
+  }
```

Comment:
again, I'd say just go for `meta.eventTime` unless you're dealing with nested properties.

get/hasProperty are useful in the same way [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is, but that is not available in node 12